functions/utils/date-utils.ts

```diff
@@ -60,6 +60,13 @@ const parseTime = (americanTime: string) => {
   const offset = americanTime.toUpperCase().endsWith('PM') ? PM_OFFSET : 0;
   const trimmedTime = americanTime.slice(0, americanTime.length - 2);
   const [hoursString, minutesString] = trimmedTime.split(':');
+
+  // handle exception cases of the midnight and noon hours
+  if (americanTime.startsWith('12')) {
```

Comment:
Fixed it and added test, I didn't realized that you fixed it too. Could you update branch once more?